nilearn/_utils/tests/test_data_gen.py

```diff
@@ -0,0 +1,116 @@
+import numpy as np
+import pytest
+from nilearn._utils.data_gen import (
+    generate_fake_fmri,
+    generate_labeled_regions,
+    generate_maps,
+    generate_regions_ts,
+)
+from nilearn.image import get_data
+
+
+@pytest.mark.parametrize("window", ["boxcar", "hamming"])
+def test_generate_regions_ts(window):
+    """Minimal testing of generate_regions_ts()."""
+    n_voxels = 50
+    n_regions = 10
+
+    # no overlap
+    regions = generate_regions_ts(
+        n_voxels, n_regions, overlap=0, window=window
+    )
+
+    assert regions.shape == (n_regions, n_voxels)
+    # check no overlap
+    np.testing.assert_array_less(
+        (regions > 0).sum(axis=0) - 0.1, np.ones(regions.shape[1])
+    )
+    # check: a region everywhere
+    np.testing.assert_array_less(
+        np.zeros(regions.shape[1]), (regions > 0).sum(axis=0)
+    )
+
```

Comment:
I would split the test here: `test_generate_regions_ts_no_overlap` and `test_generate_regions_ts_overlap`.

Reply:
done